admin/controllers/administrator.js

```diff
@@ -0,0 +1,104 @@
+const Settings = require('../models/setting')
+const SettingsLog = require('../models/setting-log')
+const settingsErrorMessages = require('../lib/errors/settings')
+const settingsValidator = require('../lib/validator/settings-validator')
+const config = require('../config')
+
+const getAdministration = async (req, res, next) => {
+  res.locals.pageTitle = 'MTC Administration Homepage'
+
+  try {
+    req.breadcrumbs(res.locals.pageTitle)
+    res.render('administrator/administration-home', {
+      breadcrumbs: req.breadcrumbs()
+    })
+  } catch (error) {
+    next(error)
+  }
+}
+
+const getUpdateTiming = async (req, res, next) => {
+  res.locals.pageTitle = 'Check settings'
+  let settings
+  const successfulPost = req.params.status || false
+
+  try {
+    const settingsRecord = await Settings.find().exec()
+    if (settingsRecord[0]) {
+      settings = settingsRecord[0]
+    } else {
+      settings = {
+        'questionTimeLimit': config.QUESTION_TIME_LIMIT,
+        'loadingTimeLimit': config.TIME_BETWEEN_QUESTIONS
+      }
+    }
+  } catch (error) {
+    return next(error)
+  }
+
+  try {
+    req.breadcrumbs(res.locals.pageTitle)
+    res.render('administrator/check-settings', {
+      settings: settings,
+      successfulPost: successfulPost,
+      breadcrumbs: req.breadcrumbs()
+    })
+  } catch (error) {
+    next(error)
+  }
+}
+
+const setUpdateTiming = async (req, res, next) => {
+  res.locals.pageTitle = 'Check settings'
+  let settings
+  let status
+
+  const settingsRecord = await Settings.find().exec()
+  if (settingsRecord[0]) {
+    settings = settingsRecord[0]
+  } else {
+    settings = new Settings()
+  }
+  settings.questionTimeLimit = Math.round(req.body.questionTimeLimit * 100) / 100
```

Comment:
Do we need to validate that the user has provided something that looks like a number before assigning it?